lnwire/extra_bytes.go

```diff
@@ -27,7 +33,13 @@ func (e *ExtraOpaqueData) Encode(w io.Writer) error {
 
 // Decode attempts to unpack the raw bytes encoded in the passed io.Reader as a
 // set of extra opaque data.
-func (e *ExtraOpaqueData) Decode(r io.Reader) error {
+func (e *ExtraOpaqueData) Decode(r io.Reader, pver uint32) error {
+	// Only if we're using the modern protocl version will we attempt to
+	// keep on decoding past the end of the "main message".
+	if pver != ProtocolVersionTLV {
```

Comment:
Can you point to the place in the migrations where this was a problem? I'm trying to understand why this is neccassary.

This also restates that the migrations shouldn't be depending on code that might change from underneath them. I would prefer rather than this change, copying the legacy `lnwire` encoding into the migrations, so we won't have to worry about this later.